File: infullmvp/src/main/java/com/infullmobile/android/infullmvp/PresentedDialogView.java
package com.infullmobile.android.infullmvp;

import android.app.Dialog;
import android.content.Context;
import android.support.annotation.IdRes;
import android.support.annotation.NonNull;
import android.view.View;

public abstract class PresentedDialogView<PresenterType extends Presenter>
        extends PresentedView<PresenterType, Dialog> {

    private Dialog boundDialog;

    @Override
    protected void bindUiElements(@NonNull Dialog dialog, @NonNull PresenterType presenter) {
        boundDialog = dialog;
        boundPresenter = presenter;
        bindViews();
        onViewsBound();
    }

    @NonNull
    @Override
    public Context getContext() {
        throwIfUnbound();
        return boundDialog.getContext();
    }

    @NonNull
    protected Dialog getBoundDialog() {
        throwIfUnbound();
        return boundDialog;
    }

    @Override
    @SuppressWarnings("unchecked")
    @edu.umd.cs.findbugs.annotations.SuppressWarnings("UWF_FIELD_NOT_INITIALIZED_IN_CONSTRUCTOR")
    protected <ViewType extends View> ViewType findView(@IdRes final int resourceId) {
        return (ViewType) boundDialog.findViewById(resourceId);
    }

    private void throwIfUnbound() {
        throw new IllegalStateException("This view should be bound to dialog first");
    }
}

File: infullmvp-kotlin/src/main/java/com/infullmobile/android/infullmvp/PresentedDialogView.kt
package com.infullmobile.android.infullmvp

import android.app.Dialog
import android.content.Context
import android.view.View

abstract class PresentedDialogView<PresenterType: Any>: ContextPresentedView<PresenterType, Dialog>() {

    private var dialog: Dialog? = null

    val boundDialog: Dialog
        get() = dialog?:throw IllegalStateException("This view must be bound to dialog")

    override val context: Context
        get() = dialog?.context?:throw IllegalStateException("This view must be bound to dialog")

    override val viewFinder: PresentedView<PresenterType, *>.(Int) -> View?
        get() = { dialog?.findViewById(it) }

    override fun bindUiElements(boundingView: Dialog, presenter: PresenterType) {
        dialog = boundingView
        boundPresenter = presenter
        onViewsBound()
    }
}
